modules/lvgl/lvgl.c

```diff
@@ -58,39 +66,33 @@ static uint8_t buf1[BUFFER_SIZE]
 		__aligned(CONFIG_LV_Z_VDB_ALIGN);
 #endif /* CONFIG_LV_Z_DOUBLE_VDB */
 
+#if ALLOC_VTILE_BUFFER
+static uint8_t mono_vtile_buf[BUFFER_SIZE]
+#ifdef CONFIG_LV_Z_VBD_CUSTOM_SECTION
+	Z_GENERIC_SECTION(.lvgl_buf)
+#endif
+		__aligned(CONFIG_LV_Z_VDB_ALIGN);
+#endif /* ALLOC_VTILE_BUFFER */
+
 #endif /* CONFIG_LV_Z_BUFFER_ALLOC_STATIC */
 
 #if CONFIG_LV_Z_LOG_LEVEL != 0
-/*
- * In LVGLv8 the signature of the logging callback has changes and it no longer
- * takes the log level as an integer argument. Instead, the log level is now
- * already part of the buffer passed to the logging callback. It's not optimal
- * but we need to live with it and parse the buffer manually to determine the
- * level and then truncate the string we actually pass to the logging framework.
- */
-static void lvgl_log(const char *buf)
+static void lvgl_log(lv_log_level_t level, const char *buf)
 {
-	/*
-	 * This is ugly and should be done in a loop or something but as it
-	 * turned out, Z_LOG()'s first argument (that specifies the log level)
-	 * cannot be an l-value...
-	 *
-	 * We also assume lvgl is sane and always supplies the level string.
-	 */
-	switch (buf[1]) {
-	case 'E':
+	switch (level) {
+	case LV_LOG_LEVEL_ERROR:
 		LOG_ERR("%s", buf + strlen("[Error] "));
```

Comment:
Since you're touching this, use compile time data instead of runtime data.

```suggestion
		LOG_ERR("%s", buf + (sizeof("[Error] ") - 1));
```

Reply:
Good idea 👍

Reply:
Done.